tests/Functional/Container.php

```diff
@@ -84,9 +113,72 @@ private function loadCmfContentType()
 
     private function loadSymfonyForm()
     {
-        $this['symfony.form_factory'] = function ($container) {
+        $this['symfony.form_factory'] = function () {
             return Forms::createFormFactoryBuilder()
                 ->getFormFactory();
         };
     }
+
+    private function loadDoctrineDbal()
+    {
+        $this['dbal.connection'] = function () {
+            return DriverManager::getConnection([
+                'driver'    => 'pdo_sqlite',
+                'path' => $this['config']['db_path']
+            ]);
+        };
+    }
+
+    private function loadPhpcrOdm()
+    {
+        $this['doctrine_phpcr.document_manager'] = function ($container) {
+
+            $registerNodeTypes = false;
+
+            // automatically setup the schema if the db doesn't exist yet.
+            if (!file_exists($container['config']['db_path'])) {
+                $connection = $container['dbal.connection'];
+
+                $schema = new RepositorySchema();
+                foreach ($schema->toSql($connection->getDatabasePlatform()) as $sql) {
+                    $connection->exec($sql);
+                }
+
+                $registerNodeTypes = true;
+            }
+
+            // register the phpcr session
+            $factory = new RepositoryFactoryDoctrineDBAL();
+            $repository = $factory->getRepository([
+                'jackalope.doctrine_dbal_connection' => $container['dbal.connection']
+            ]);
+            $session = $repository->login(new SimpleCredentials(null, null), 'default');
+
+            if ($registerNodeTypes) {
+                $typeRegistrator = new NodeTypeRegistrator();
+                $typeRegistrator->registerNodeTypes($session);
+            }
+
+            // content type driver
+            $contentTypeDriver = new ContentTypeDriver(
+                $container['cmf_content_type.registry.field'],
+                $container['cmf_content_type.registry.mapping']
+            );
+
+            // annotation driver
+            $reader = new AnnotationReader();
+            $annotationDriver = new AnnotationDriver($reader, [
+                __DIR__ . '/../../vendor/doctrine/phpcr-odm/lib/Doctrine/ODM/PHPCR/Document',
+            ]);
+            $chain = new MappingDriverChain();
+            $chain->addDriver($contentTypeDriver, 'Symfony');
```

Comment:
The second argument is a namespace prefix and it is mandatory. We will need to register the content type namespaces before this happens and add a driver for each unique namespace (?) /cc @dbu